File: C/test/test_finalPositionOfSnake.c

#ifdef TEST

#include "unity.h"

#include "finalPositionOfSnake.h"

void setUp(void)
{
}

void tearDown(void)
{
}

void test_finalPositionOfSnake_1(void)
{
	int n = 2;
	char* commands[] = {"RIGHT","DOWN"};
	int expect= 3;

	TEST_ASSERT_EQUAL(expect, finalPositionOfSnake(n, commands, sizeof(commands)/sizeof(commands[0])));
}

void test_finalPositionOfSnake_2(void)
{
	int n = 3;
	char* commands[] = {"DOWN","RIGHT","UP"};
	int expect= 1;

	TEST_ASSERT_EQUAL(expect, finalPositionOfSnake(n, commands, sizeof(commands)/sizeof(commands[0])));
}

#endif // TEST

File: Ruby/test/test_final_position_of_snake.rb
require_relative "test_helper"
require_relative "../solution/final_position_of_snake"

test_cases = [
  [[2, ["RIGHT", "DOWN"]], 3],
  [[3, ["DOWN", "RIGHT", "UP"]], 1],
]

run_tests(test_cases, method(:final_position_of_snake))
